File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.java
class Solution {
    public int solution(int left, int right) {
        int answer = 0;
        for (int n = left; n < right + 1; n++) {
            if (getDivisorCount(n) % 2 == 0) {
                answer += n;
            } else {
                answer -= n;
            }
        }
        return answer;
    }
    
    public int getDivisorCount(int n) {
        int cnt = 0;
        for (int i = 1; i < (int) Math.sqrt(n) + 1; i++) {
            if (n % i == 0) {
                cnt += 1;
                if (i * i != n) {
                    cnt += 1;
                }
            }
        }
        return cnt;
    }
}

File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.py
def get_divisor_count(n):
    cnt = 0
    for i in range(1, int(n ** 0.5) + 1):
        if n % i == 0:
            cnt += 1
            if i ** 2 != n:
                cnt += 1
    return cnt

def solution(left, right):
    answer = 0
    for n in range(left, right + 1):
        if get_divisor_count(n) % 2 == 0:
            answer += n
        else:
            answer -= n
    return answer